### PR TITLE
Log burst metric headers

### PR DESCRIPTION
**Pull request by rcaseymsft**
# Type of PR

- [ ] Documentation changes
- [ ] Code changes
- [x] Test changes
- [ ] CI-CD changes
- [ ] GitHub Template changes

## PR Checklist

- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.
- [ ] My code follows the code style of this project.
- [ ] I ran lint checks locally prior to submission.
- [ ] Have you checked to ensure there aren't other open Pull Requests for the same update/change?

## Purpose of PR

Add X-Load-Feedback header to logged information.

## Does this introduce a breaking change

- [ ] YES
- [x] NO

## Validation

- [x] Unit tests updated and ran successfully
- [ ] Update documentation or issue referenced above

## Issues Closed or Referenced

- Closes https://github.com/retaildevcrews/wcnp/issues/743

### CODE REVIEW

**Comment by joaquinrz** [comment]
@rcaseymsft I think we should have a design review for this PR. I am not entirely sure if we need to have burst metrics yaml inside the loderunner repo (I am thinking in terms of maintainability and reuse). It might be the case that creating a burst metrics helm chart that we can reference be appropriate for this scenario. But I think it's best to have that discussion with the overall team. @jofultz thoughts?

**Comment by kforeverisback** [comment]
> the

I was sort of hung on that as well. Since burst metrics needs to be deployed for Loderunner to pick up the header, we wanted to keep everything contained here in this repo, which would be consistent with a local cluster.

My concern is the wasm binary we are using. The binary is not committed in this repo, rather we're downloading it from [ngsa-asb repo](https://github.com/retaildevcrews/ngsa-asb/blob/main/wasm/burst_header.wasm).

We have couple of options we can use to acquire the binary.
We can discuss that in a design review session

**Review by jofultz — CHANGES_REQUESTED** [review]
@rcaseymsft, I'm blocking so that we can discuss.  I don't think including the deployment in this repo is ideal, but wanted to discuss your perspective and see if it changes my thoughts on it.  If we do decide that it is the best approach, we'll need to update the readme as well.

**Comment by kforeverisback** [comment]
@jofultz We had a small discussion yesterday. And according to the discussion we added descriptions and instructions on how to use Burst Metrics header in this environment.

Also, we've changed the wasm binary URL from `main` branch to a specific commit so that, updating the burst wasm binary doesn't break the dev environment.

File: Makefile [hunk 1]
```diff
@@ -32,11 +34,46 @@ create : delete
...
+
+	# patch ngsa-memory
+	@# this will create a new ngsa deployment and terminate the old one
+	@kubectl patch deployment -n ngsa ngsa-memory -p '{"spec":{"template":{"metadata":{"annotations":{"sidecar.istio.io/userVolume":"[{\"name\":\"wasmfilters-dir\",\"configMap\": {\"name\": \"burst-wasm-filter\"}}]","sidecar.istio.io/userVolumeMount":"[{\"mountPath\":\"/var/local/lib/wasm-filters\",\"name\":\"wasmfilters-dir\"}]"}}}}}'
+
+	# wait on ngsa-memory pods
+	kubectl wait po -n ngsa --all --for condition=ready --timeout=60s
+
+	# turn the wasm filter on
+	@kubectl apply -f deploy/burst/ngsa-filter-sidecar.yaml
```
Comment (sivamu) [inline Makefile:68]: Same thing here, maybe just apply the remote file? I don't see any differences of this sidecar file and the one in istio so the istio copy should be applicable  as-is

**Review by sivamu — CHANGES_REQUESTED** [review]
See review comments

File: Makefile [hunk 2]
```diff
@@ -32,11 +34,46 @@ create : delete
 	@sleep 5
 	@kubectl wait pod -A --all --for condition=ready --timeout=60s
 
-deploy :
+	# install istio
+	@/usr/local/istio/bin/istioctl install --set profile=demo -y
+
+deploy-burst: deploy-ngsa
+	# deploy burst metrics
+	@kubectl apply -f deploy/burst/burst-dev.yaml
```
[the remark this answers is not in the document]
Comment (kforeverisback) [reply Makefile:42]: @sivamu Couple of reasons:
- We wanted to be consistent with other deployments (e.g. ngsa, fluentbit etc).
- With a local copy its self-sufficient local dev env.
- Other repo's burst metrics yaml could change depending on the requirements.
  - For example, burst yaml in istio repo is using local version of the image
  - NGSA-Asb's burst dev has all checkov labels and tmp directory mount

[the remark this answers is not in the document]
Comment (sivamu) [reply Makefile:42]: makes sense. this is something that joseph brought up in his email, where we have copies of ngsa in multiple repos and we would need to revisit the organization for all repos sometime later. for the time being, let's go ahead with this approach.

**Review by sivamu — APPROVED** [review]

**Review by jofultz — APPROVED** [review]
Approving based on team review and input.

**Pull request merged**